pkg/registry/core/service/storage/rest.go

```diff
@@ -845,8 +872,46 @@ func (al *RESTAllocStuff) initIPFamilyFields(oldService, service *api.Service) e
 		return nil // nothing more to do.
 	}
 
+	// Update only prep work.
+	if oldService != nil {
+		np := service.Spec.IPFamilyPolicy
+
+		// If they didn't specify policy, or specified anything but
+		// single-stack AND they reduced these fields, it's an error.  They
+		// need to specify policy.
+		if np == nil || *np != api.IPFamilyPolicySingleStack {
```

Comment:
Ok. `on update policy must be set` That means on Post/Patch/etc the user must either `NOT touch it` or `*DELIBERATELY change it*`. that is the agreement right? so:
```go
if !isUpdated(old, new) /* checks if old != nil *{
 // split the func in two: one for create and one for update
 // call create logic
}
// ASSUMING THAT HEADLESS AND EXTERNAL NAME HAS BEEN HANDLED
// the rest of function is for update case only
 np := service.Spec.IPFamilyPolicy
 if np == nil{ 
   // error immutable once set.
 }
 // single stack case
 if np == api.IPFamilyPolicySingleStack /*we don't care what old was here*/{
 // the below should never fail since it is 
 //1) it is on update path  [i.e.,must have something set prior]
 //2) !ExternalName
 //3) !Headless
 service.Spec.ClusterIPs = service.Spec.ClusterIPs[:1]
 service.Spec.IPFamilies = service.Spec.IPFamilies[:1]
 
 // return here
 }
 // at this stage we know np is not nil - we also know it is not SingleStack
if reducedClusterIPs(oldService, service) {...}
if reducedIPFamilies(oldService, service) {...}
if len(el) > 0 {.. return errors ...}
return here
}
// the only remaining logic is testing immutability of pre allocated ClusterIPs. both are immutable once set.
```